https://salsa.debian.org/debian/dia/-/blob/debian/0.98+git20250827-2/debian/patches/revert_xpm_loader_replacement.patch
See https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/169
This patch reverts upstream commit 1ba72d126055130474b4ce589b1f3002972ba490
"general: use xpm-pixbuf to load xpms".
Use loader from GdkPixbuf as long as available, instead of the fork xpm-pixbuf
--- a/app/dia-colour-area.c
+++ b/app/dia-colour-area.c
@@ -22,8 +22,6 @@
 
 #include <stdio.h>
 
-#include <xpm-pixbuf.h>
-
 #include "attributes.h"
 #include "persistence.h"
 
@@ -297,8 +295,8 @@ dia_colour_area_class_init (DiaColourAreaClass *class)
 static void
 dia_colour_area_init (DiaColourArea *self)
 {
-  self->reset = xpm_pixbuf_load (default_xpm);
-  self->swap = xpm_pixbuf_load (swap_xpm);
+  self->reset = gdk_pixbuf_new_from_xpm_data (default_xpm);
+  self->swap = gdk_pixbuf_new_from_xpm_data (swap_xpm);
 
   gtk_widget_set_events (GTK_WIDGET (self), GDK_BUTTON_PRESS_MASK);
 }
--- a/app/dia-page-layout.c
+++ b/app/dia-page-layout.c
@@ -25,8 +25,6 @@
 
 #include <glib/gi18n-lib.h>
 
-#include <xpm-pixbuf.h>
-
 #include "dia-page-layout.h"
 
 #include "pixmaps/portrait.xpm"
@@ -148,7 +146,7 @@ dia_page_layout_init (DiaPageLayout *self)
 
   self->orient_portrait = gtk_radio_button_new (NULL);
 
-  wid = gtk_image_new_from_pixbuf (xpm_pixbuf_load (portrait_xpm));
+  wid = gtk_image_new_from_pixbuf (gdk_pixbuf_new_from_xpm_data (portrait_xpm));
   gtk_container_add (GTK_CONTAINER (self->orient_portrait), wid);
   gtk_widget_show (wid);
 
@@ -157,7 +155,7 @@ dia_page_layout_init (DiaPageLayout *self)
 
   self->orient_landscape = gtk_radio_button_new (
         gtk_radio_button_get_group (GTK_RADIO_BUTTON (self->orient_portrait)));
-  wid = gtk_image_new_from_pixbuf (xpm_pixbuf_load (landscape_xpm));
+  wid = gtk_image_new_from_pixbuf (gdk_pixbuf_new_from_xpm_data (landscape_xpm));
   gtk_container_add (GTK_CONTAINER (self->orient_landscape), wid);
   gtk_widget_show (wid);
 
--- a/app/navigation.c
+++ b/app/navigation.c
@@ -24,8 +24,6 @@
 
 #include <gtk/gtk.h>
 
-#include <xpm-pixbuf.h>
-
 #include "diagram.h"
 #include "display.h"
 #include "renderer/diacairo.h"
@@ -489,7 +487,7 @@ navigation_popup_new (DDisplay *ddisp)
                     "released", G_CALLBACK (on_button_navigation_popup_released),
                     NULL);
 
-  pixbuf = xpm_pixbuf_load (nav_xpm);
+  pixbuf = gdk_pixbuf_new_from_xpm_data (nav_xpm);
 
   image = gtk_image_new_from_pixbuf (pixbuf);
 
--- a/app/sheet-editor/sheet-editor-button.c
+++ b/app/sheet-editor/sheet-editor-button.c
@@ -24,8 +24,6 @@
 
 #include <glib/gi18n-lib.h>
 
-#include <xpm-pixbuf.h>
-
 #include "sheet-editor-button.h"
 #include "widgets.h"
 #include "message.h"
@@ -79,7 +77,7 @@ get_object_pixbuf (SheetObject *so)
     if (g_str_has_prefix ((char *) so->pixmap, "res:")) {
       pixbuf = pixbuf_from_resource (((char *) so->pixmap) + 4);
     } else {
-      pixbuf = xpm_pixbuf_load (so->pixmap);
+      pixbuf = gdk_pixbuf_new_from_xpm_data ((const char **) so->pixmap);
     }
   } else {
     if (so->pixmap_file != NULL) {
--- a/app/toolbox.c
+++ b/app/toolbox.c
@@ -21,9 +21,6 @@
 #include <glib/gi18n-lib.h>
 
 #include <gtk/gtk.h>
-
-#include <xpm-pixbuf.h>
-
 #include "gtkwrapbox.h"
 #include "gtkhwrapbox.h"
 
@@ -305,7 +302,7 @@ fill_sheet_wbox(Sheet *sheet)
       if (g_str_has_prefix ((const char *) sheet_obj->pixmap, "res:")) {
         pixbuf = pixbuf_from_resource ((const char *) sheet_obj->pixmap + 4);
       } else {
-        pixbuf = xpm_pixbuf_load (sheet_obj->pixmap);
+        pixbuf = gdk_pixbuf_new_from_xpm_data (sheet_obj->pixmap);
         if (pixbuf == NULL) {
           g_warning ("Problem loading icon for ‘%s’", sheet_obj->description);
           pixbuf = gtk_icon_theme_load_icon (gtk_icon_theme_get_default (),
@@ -334,7 +331,7 @@ fill_sheet_wbox(Sheet *sheet)
           pixbuf = cropped;
         }
       } else {
-        pixbuf = xpm_pixbuf_load (missing);
+        pixbuf = gdk_pixbuf_new_from_xpm_data (missing);
 
         message_warning ("failed to load icon for file\n %s\n cause=%s",
                          sheet_obj->pixmap_file,
@@ -342,8 +339,8 @@ fill_sheet_wbox(Sheet *sheet)
       }
     } else {
       DiaObjectType *type;
-      type = object_get_type (sheet_obj->object_type);
-      pixbuf = dia_object_type_get_icon (type);
+      type = object_get_type(sheet_obj->object_type);
+      pixbuf = gdk_pixbuf_new_from_xpm_data (type->pixmap);
     }
 
     if (pixbuf) {
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -230,7 +230,6 @@ libdia_deps = [
     libm_dep,
     gmodule_dep,
     graphene_dep,
-    xpm_dep,
 ]
 
 libdia_inc = include_directories('.')
--- a/lib/object.c
+++ b/lib/object.c
@@ -23,8 +23,6 @@
 #include <stdio.h>
 #include <string.h>
 
-#include <xpm-pixbuf.h>
-
 #define _DIA_OBJECT_BUILD 1
 #include "object.h"
 #include "diagramdata.h" /* for Layer */
@@ -1425,6 +1423,14 @@ dia_object_type_get_icon (const DiaObjectType *type)
 
   if (g_str_has_prefix ((char *) icon_data, "res:")) {
     pixbuf = pixbuf_from_resource (((char *) icon_data) + 4);
+  } else if (icon_data && strncmp ((char *) icon_data, "GdkP", 4) == 0) {
+    /* GTK3: We will remove this in Dia 2.0 but is retained to maintain
+     * compatability with 0.97.3 objects */
+    g_warning ("Object '%s' has an inline icon, this is deprecated",
+               type->name);
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+    pixbuf = gdk_pixbuf_new_from_inline (-1, (guint8 *) icon_data, TRUE, NULL);
+    G_GNUC_END_IGNORE_DEPRECATIONS
   } else if (type->pixmap_file != NULL) {
     GError *error = NULL;
     pixbuf = gdk_pixbuf_new_from_file (type->pixmap_file, &error);
@@ -1433,7 +1439,8 @@ dia_object_type_get_icon (const DiaObjectType *type)
       g_clear_error (&error);
     }
   } else {
-    pixbuf = xpm_pixbuf_load (icon_data);
+    const char **pixmap_data = icon_data;
+    pixbuf = gdk_pixbuf_new_from_xpm_data (pixmap_data);
   }
 
   return pixbuf;
--- a/meson.build
+++ b/meson.build
@@ -22,7 +22,6 @@ libxml_dep  = dependency('libxml-2.0', version: '>= 2.14.0')
 gmodule_dep = dependency('gmodule-2.0')
 libcairo_dep = dependency('cairo')
 graphene_dep = dependency('graphene-1.0', version: '>= 1.10')
-xpm_dep = dependency('xpm-pixbuf')
 
 # Not required since not all platforms ship a separate libm.
 libm_dep = cc.find_library('m', required: false)
